### PR TITLE
Implements geometric mean initialization

### PR DESCRIPTION
**Pull request by adefazio**
An implementation of the geometric mean initialization scheme from my recent preprint https://arxiv.org/abs/1906.04267. This is a small addition that follows the code for other initialization schemes closely.

### CODE REVIEW

**Comment by soumith** [comment]
We don't think this method should be in pytorch core, as opposed to your own personal repository or something like https://github.com/pytorch/contrib , at least not yet in time.

Our reservation is that we want to include methods that the community uses as a standard, or else the code maintenance problem balloons up for us.
We do show discretion based on what the paper shows as evidence, for example BatchNorm was included within weeks of it's publish date (in Torch).
In terms of rejected methods, we've rejected newly minted papers such as Swish ( #3260 , #3182 ), Yellowfin ( https://github.com/pytorch/pytorch/issues/1960 ) and many others, and rightly so, these haven't become standardized in the community (like LSTM / Transformer / BatchNorm).

If you have a differing opinion, let us know why, and we can re-think.

tl;dr: The paper doesn't show evidence that makes it a method that has obvious long-term success. If the paper does have long-term success in the field we will include it

**Comment by soumith** [comment]
Opened https://github.com/pytorch/pytorch/pull/23513 so that we give better guidance that an Issue should be opened first before putting in the effort to push a PR
